core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -286,6 +286,29 @@ private[spark] class TaskSchedulerImpl(
     }
   }
 
+  /**
+   * SPARK-25250: Whenever any Result Task gets successfully completed, we simply mark the
+   * corresponding partition id as completed in all attempts for that particular stage. As a
+   * result, we do not see any Killed tasks due to TaskCommitDenied Exceptions showing up
+   * in the UI.
+   */
+  override def markPartitionIdAsCompletedAndKillCorrespondingTaskAttempts(
```

Comment:
Doesn't this logic overlap with `killAllTaskAttempts`? should it reuse that logic? I understand it does something a little different, and I don't know this code well, but seems like there are related but separate implementations of something similar here.

Reply:
As far as I understand the code, `killAllTaskAttempts` kills all the running tasks for a particular stage whereas `markPartitionIdAsCompletedAndKillCorrespondingTaskAttempts` kills all running tasks for all stages and attempts working on a particular partition that has already been marked as completed by one of the previously running tasks for that corresponding partition. So the logic is different for both the cases, but we can modify the code to have one fixed method for performing both these tasks. Let me know what you think!